bindgen/ir/function.rs

```diff
@@ -728,6 +732,16 @@ impl ClangSubItemParser for Function {
 
         let function =
             Self::new(name, mangled_name, sig, comment, kind, linkage);
+
+        if matches!(linkage, Linkage::Internal) &&
+            context.options().generate_extern_functions.is_true()
+        {
+            match CItem::from_function(&function, context) {
+                Ok(c_item) => context.c_items.push(c_item),
```

Comment:
So one issue with putting this in the `BindgenContext` is that it won't respect any of the filtering options. I think we should do this on codegen, so adding a field to `CodegenResult`, and dealing with those there. Is there any reason that wouldn't work?